charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -20,3 +20,163 @@ spec:
+                          type: string
+                        size:
+                          type: string
+                        createPVC:
+                          type: boolean
+                      oneOf:
+                      - required:
+                        - claimName
+                      - required:
+                        - storageClass
+                        - createPVC
+                        properties:
+                          createPVC:
+                            enum:
+                            - true
+                  oneOf:
+                    # note: current problem - is there a way to enforce only hdfs is set?
+                    # note: for above, maybe move this `oneOf` to the parent column and verify all fields are set?
+                  - required:
+                    - hdfs
+                    properties:
+                      type:
+                        enum:
+                        - hdfs
+                  - required:
+                    - s3
+                    properties:
+                      type:
+                        enum:
+                        - s3
+                  - required:
+                    - azure
+                    properties:
+                      type:
+                        enum:
+                        - azure
+                  - required:
+                    - sharedPVC
+                    properties:
+                      type:
+                        enum:
+                        - sharedPVC
+            hadoop:
+              type: object
+              properties:
+                spec:
+                  type: object
+                  properties:
+                    hdfs:
+                      type: object
+                      properties:
+                        datanode:
+                          type: object
+                          properties:
+                            # note: should this be added in this PR (not including resource limits)?
+                            replicas:
+                              type: integer
+                              minimum: 0
+                            storage:
+                              type: object
+                              properties:
+                                class:
+                                  type: string
+                                size:
+                                  type: string
+                        namenode:
+                          type: object
+                          properties:
+                            replicas:
+                              type: integer
+                              minimum: 0
+                            storage:
+                              type: object
+                              properties:
+                                class:
+                                  type: string
+                                size:
+                                  type: string
```

Comment:
I could potentially add a pattern for the `size` field that checks whether or not the value is properly formatted.